library/std/src/sys/uefi/common.rs

```diff
@@ -0,0 +1,477 @@
+            const_io_error!(ErrorKind::HostUnreachable, "EFI_NO_RESPONSE")
+        }
+        Status::NOT_FOUND => const_io_error!(ErrorKind::NotFound, "EFI_NOT_FOUND"),
+        Status::NOT_READY => const_io_error!(ErrorKind::ResourceBusy, "EFI_NOT_READY"),
+        Status::NOT_STARTED => const_io_error!(ErrorKind::Other, "EFI_NOT_STARTED"),
+        Status::OUT_OF_RESOURCES => {
+            const_io_error!(ErrorKind::OutOfMemory, "EFI_OUT_OF_RESOURCES")
+        }
+        Status::PROTOCOL_ERROR => {
+            const_io_error!(ErrorKind::Other, "EFI_PROTOCOL_ERROR")
+        }
+        Status::PROTOCOL_UNREACHABLE => {
+            const_io_error!(ErrorKind::Other, "EFI_PROTOCOL_UNREACHABLE")
+        }
+        Status::SECURITY_VIOLATION => {
+            const_io_error!(ErrorKind::PermissionDenied, "EFI_SECURITY_VIOLATION")
+        }
+        Status::TFTP_ERROR => const_io_error!(ErrorKind::Other, "EFI_TFTP_ERROR"),
+        Status::TIMEOUT => const_io_error!(ErrorKind::TimedOut, "EFI_TIMEOUT"),
+        Status::UNSUPPORTED => {
+            const_io_error!(ErrorKind::Unsupported, "EFI_UNSUPPORTED")
+        }
+        Status::VOLUME_FULL => {
+            const_io_error!(ErrorKind::StorageFull, "EFI_VOLUME_FULL")
+        }
+        Status::VOLUME_CORRUPTED => {
+            const_io_error!(ErrorKind::Other, "EFI_VOLUME_CORRUPTED")
+        }
+        Status::WRITE_PROTECTED => {
+            const_io_error!(ErrorKind::ReadOnlyFilesystem, "EFI_WRITE_PROTECTED")
+        }
+        _ => io::Error::new(ErrorKind::Uncategorized, format!("Status: {}", s.as_usize())),
+    }
+}
+
+pub fn install_protocol<T>(
+    handle: &mut r_efi::efi::Handle,
+    mut guid: r_efi::efi::Guid,
+    interface: &mut T,
+) -> io::Result<()> {
+    let boot_services = uefi::env::get_boot_services().ok_or(BOOT_SERVICES_ERROR)?;
+    let r = unsafe {
+        ((*boot_services.as_ptr()).install_protocol_interface)(
+            handle,
+            &mut guid,
+            r_efi::efi::NATIVE_INTERFACE,
+            interface as *mut T as *mut crate::ffi::c_void,
+        )
+    };
+    if r.is_error() { Err(status_to_io_error(r)) } else { Ok(()) }
+}
+
+pub fn uninstall_protocol<T>(
+    handle: r_efi::efi::Handle,
+    mut guid: r_efi::efi::Guid,
+    interface: &mut T,
+) -> io::Result<()> {
+    let boot_services = uefi::env::get_boot_services().ok_or(BOOT_SERVICES_ERROR)?;
+    let r = unsafe {
+        ((*boot_services.as_ptr()).uninstall_protocol_interface)(
+            handle,
+            &mut guid,
+            interface as *mut T as *mut crate::ffi::c_void,
+        )
+    };
+    if r.is_error() { Err(status_to_io_error(r)) } else { Ok(()) }
+}
+
+// A Helper trait for `ProtocolWrapper<P>`.
+pub trait Protocol {
+    const PROTOCOL_GUID: Guid;
+}
+
+// A wrapper for creating protocols from Rust in Local scope. Uninstalls the protocol on Drop.
+// Ideally, the Protocol should not move in memory once installed. However, not sure if I need to
```

Comment:
You do not need to -- this isn't exposed to users anyway so it wouldn't matter that much anyway.

Please add questions like this as a github issue comment rather than comments in the source, because it's easy for me to miss them otherwise.